src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1576,22 +1576,28 @@ fn resolution_failure(
                     };
                     // See if this was a module: `[path]` or `[std::io::nope]`
                     if let Some(module) = last_found_module {
-                        let module_name = collector.cx.tcx.item_name(module);
-                        let note = format!(
-                            "the module `{}` contains no item named `{}`",
-                            module_name, unresolved
-                        );
+                        let note = if partial_res.is_some() {
+                            let module_name = collector.cx.tcx.item_name(module);
+                            format!(
+                                "the module `{}` contains no item named `{}`",
+                                module_name, unresolved
+                            )
+                        } else {
+                            format!("there is no item named `{}` in scope", unresolved)
```

Comment:
This is a little wordy; should I change it to this? I'm not sure if it's less clear though.

```suggestion
                            format!("`{}` is not in scope", unresolved)
```

Reply:
I think
```suggestion
                            format!("no item named `{}` in scope", unresolved)
```
would be equally clear and less verbose. 'x is not in scope' works but seems a little _too_ concise, IDK. I'd be fine with either.

Reply:
I at first went for "{} is not in scope" but then I realized yours was clearer :)

The shorter one "looks" nicer, but it makes it sound like there definitely is something named "{}", you just need to import it, when in fact it's possible it doesn't exist. Maybe when we get access to `rustc_resolve`'s fancy errors we can change the message based on whether there is something named "{}" that you could import.